staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -684,6 +684,9 @@ func TestRoundTripSocks5AndNewConnection(t *testing.T) {
 							t.Fatalf("unexpected proxy user: %v", authUser)
 						}
 					}
+
+					// The channel must be closed before any of the connections are closed
+					close(closed)
```

Comment:
should we move this up to just after the `defer conn.Close()`? not doing this in a defer seems leaky/fragile

Reply:
I was worried about that too,but seems from that `defer` to the end  of the test, there are only `t.Fatalf` statements